.github/workflows/build.yml

```diff
@@ -32,9 +34,21 @@ jobs:
           cache-read-only: true
 
       - name: Build with Gradle
-        run: ./gradlew clean ktlintCheck build koverLog koverHtmlReport
-        env:
-          JAVA_OPTS: "-Xmx8g -Dfile.encoding=UTF-8 -Djava.awt.headless=true -Dkotlin.daemon.jvm.options=-Xmx6g"
+        run: |-
+          ./gradlew clean ktlintCheck build koverLog koverHtmlReport
+          ./gradlew :kotlin-sdk-core:publishToMavenLocal :kotlin-sdk-client:publishToMavenLocal :kotlin-sdk-server:publishToMavenLocal
+
+      - name: Build Kotlin-MCP-Client Sample
+        working-directory: ./samples/kotlin-mcp-client
+        run: ./../../gradlew clean build
+
+      - name: Build Kotlin-MCP-Server Sample
+        working-directory: ./samples/kotlin-mcp-server
+        run: ./../../gradlew clean build
+
+      - name: Build Weather-Stdio-Server Sample
+        working-directory: ./samples/weather-stdio-server
+        run: ./../../gradlew clean build
```

Comment:
you can simplify this by using the following:
`./gradlew kotlin-mcp-client:build kotlin-mcp-server:build weather-stdio-server:build`